readthedocs/analytics/models.py

```diff
@@ -28,65 +51,94 @@ class PageView(models.Model):
         related_name='page_views',
         on_delete=models.CASCADE,
     )
+    # NOTE: this could potentially be removed,
+    # since isn't being used and not all page
+    # views (404s) are attached to a version.
     version = models.ForeignKey(
         Version,
         verbose_name=_('Version'),
         related_name='page_views',
         on_delete=models.CASCADE,
+        null=True,
+    )
+    path = models.CharField(
+        max_length=4096,
+        help_text=_("Path relative to the version."),
+    )
+    full_path = models.CharField(
+        max_length=4096,
+        help_text=_("Full path including the version and language parts."),
+        null=True,
+        blank=True,
     )
-    path = models.CharField(max_length=4096)
     view_count = models.PositiveIntegerField(default=0)
     date = models.DateField(default=datetime.date.today, db_index=True)
+    status = models.PositiveIntegerField(
+        default=200,
+        help_text=_("HTTP status code"),
+    )
+
+    objects = PageViewManager()
 
     class Meta:
-        unique_together = ("project", "version", "path", "date")
+        unique_together = ("project", "version", "path", "date", "status")
 
     def __str__(self):
-        return f'PageView: [{self.project.slug}:{self.version.slug}] - {self.path} for {self.date}'
+        return f"PageView: [{self.project.slug}] - {self.full_path or self.path} for {self.date}"
 
     @classmethod
-    def top_viewed_pages(cls, project, since=None, limit=10):
+    def top_viewed_pages(
+        cls, project, since=None, limit=10, status=200, per_version=False
+    ):
         """
         Returns top pages according to view counts.
 
-        Structure of returned data is compatible to make graphs.
-        Sample returned data::
-        {
-            'pages': ['index', 'config-file/v1', 'intro/import-guide'],
-            'view_counts': [150, 120, 100]
-        }
-        This data shows that `index` is the most viewed page having 150 total views,
-        followed by `config-file/v1` and `intro/import-guide` having 120 and
-        100 total page views respectively.
+        :param per_version: If `True`, group the results per version.
```

Comment:
We could change this name to something else, because now it actually means that we are going to group by `full_path` (which contains the version) instead of `path`.